diag_integral/diag_integral.F90

```diff
@@ -1522,7 +1527,40 @@ function vert_diag_integral (data, wt) result (data2)
 
 end function vert_diag_integral
 
-
+function ensemble_file_name(fname) result(updated_file_name)
+     character (len=mxch), intent(inout) :: fname
+     character (len=mxch) :: updated_file_name
+     integer :: ensemble_id_int
+     character(len=7) :: ensemble_suffix
+     character(len=2) :: ensemble_id_char 
+     integer :: i
+     !> Make sure the file name short enough ti handle adding the ensemble number
+     if (len(trim(fname)) > mxch-7) call error_mesg ('diag_integral_mod :: ensemble_file_name',  &
+          trim(fname)//" is too long and can not support adding ens_XX.  Please shorten the "//&
+          "file_name in the diag_integral_nml", FATAL)
+     !> Get the ensemble ID and convert it to a string
+          ensemble_id_int = get_ensemble_id()
+          write(ensemble_id_char,"(I0)") ensemble_id_int
+     !> Add a 0 if the ensemble is less than 10 (2 digit)
+          if (ensemble_id_int < 10) then
+               ensemble_suffix = ".ens_0"//trim(ensemble_id_char)
+          elseif (ensemble_id_int >= 10 .and. ensemble_id_int < 100) then
+               ensemble_suffix = ".ens_"//trim(ensemble_id_char)
+          else
+               call error_mesg ('diag_integral_mod',  &
+                ' Does not support ensemble sizes over 99.', FATAL)
```

Comment:
I know MAX_ENSEMBLE_SIZE is[ currently set to 100](https://github.com/NOAA-GFDL/FMS/blob/bb4e81fe3598af0069eb0ad576d6dd956275d209/coupler/ensemble_manager.F90#L35) in ensemble_manager.F90, but is this size sufficient to support the current (or anticipated) standard ensemble sizes defined in the atmos/ESM workflows of the users who want this feature implemented?

Reply:
No, but this is a conversation that is beyond the scope of this merge request.  It's also a simple enough fix here, BUT running 100 model runs with 864 cores is also beyond the limits of what we are currently allowed to run on our production system.  This is definitely a larger conversation.

Reply:
All right. Thanks for confirming that this is a resource limitation rather than a standard workload size.